### ISSUE TITLE
Run TestFX by writing test scripts in plain English text

### ISSUE THREAD

**Issue by FinancialInstituteDeveloper** [TestFX/TestFX#289]
Hi, 

This may have been raised before. 

Is there supports in TestFX to test the JavaFX Application by writing just the plain text scripts. The idea is to define the test cases in a text file and feed it to TestFX to execute the test cases automatically:

Instead of writing the robot codes using TestFX, can testers just write steps like following and run through the TestFX libraries?
1. start the application
2. input text value in (.textfield 1)
3. click submit button 
4. assert display result

Thanks,

**Comment by javier-jimenez** [comment]
You can achieve this with the Robot Framework (robotframework.org) you simply need to create a small library wrapping TestFX and calling TestFX methods inside your own keywords.

**Comment by RobzInadE** [comment]
I myself use a combination of `cucumber-java` and TestFx, giving me some sweet readable user-stories.

My tests looks something like this:

```
Given I navigate to the tab "Tab1"
And I fill in the text "blabla" in field "Field1"
When I click "Save"
Then the text "blabla" has been saved to file
```

Or another example: [Image Link](http://image.slidesharecdn.com/cucumber-141116213139-conversion-gate01/95/cucumber-bdd-27-638.jpg?cb=1416173565)

Check out https://cucumber.io/docs/reference/jvm
If you need more info on how to get TestFx and cucumber-jvm to run together, holler.

**Comment by brcolow** [comment]
Having Spock-like features would be nice, but I am going to close this as out of scope for now as we have bigger fish to fry.
